src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -58,10 +58,18 @@ public function __toString()
     /**
      * Returns the headers.
      *
+     * @param string|null $key The name of the headers to return or null to get them all
+     *
      * @return array An array of headers
      */
-    public function all()
+    public function all(/*string $key = null*/)
```

Comment:
That is a no-go for me. It introduces another of #31317 which this PR tried to fix.
Depending on the arguments it returns an array of arrays or an array of strings.

Reply:
I know it's the same as on the Messenger Envelope but that was on my todo to fix as was a pain to work with.